### PR TITLE
Halogen 1.0.0

### PR DESCRIPTION
**Pull request by cryogenian**
This is to send messages dismissing and selecting submenus from parent component

### CODE REVIEW

File: src/Halogen/Menu/Component.purs [hunk 1]
```diff
@@ -25,8 +27,8 @@ import Halogen.Menu.Submenu.Component as SUB
 
 data Query a next
   = Set (State a) next
-  | SelectSubmenu Int DET.MouseEvent next
-  | DismissSubmenu DET.MouseEvent next
+  | SelectSubmenu Int (Maybe DET.MouseEvent) next
+  | DismissSubmenu (Maybe DET.MouseEvent) next
   | HandleSubmenu (SUB.Message a) next
```
Comment (garyb) [inline src/Halogen/Menu/Component.purs]: Maybe we could add `HandleMouseEvent DET.MouseEvent (Query a)`, then in the eval it'd be something like:

``` purescript
HandleMouseEvent mbe q ->
  for_ mbe \e → liftEff do
    DEE.preventDefault $ DET.mouseEventToEvent e
    DEE.stopPropagation $ DET.mouseEventToEvent e
  eval q
```

That way the Select/Dismiss submenu stuff can have a more pleasant interface for external triggering.

Comment (cryogenian) [reply src/Halogen/Menu/Component.purs]: Okay :+1:

**Comment by cryogenian** [comment]
@garyb Does this look ok?

**Review by garyb — CHANGES_REQUESTED** [review]
I made a bit of a mistake in my suggested query, what I was getting at was having a constructor like: 

```
HandleMouseEvent Event (Query a next)
```

Defining it this way makes it clear that the inner query will always be used - as in fact you can't implement `eval` without running the inner query this way.

**Comment by cryogenian** [comment]
@garyb changed to `HandleMouseEvent Event (Query a next)`

**Review by garyb — APPROVED** [review]

**Pull request merged**